### DISCUSSION TITLE
Persist called unnecessarily

### DISCUSSION THREAD

**Discussion by Fabioni** [pmndrs/zustand#2413] [Bug report]
Persist `setItem` is called every time the global `set()` function is called even
- if there was no change at all and 
- even if there was no change inside the partialized part of the store

So there are many many unnecessarily updates to persistent storages. Since the documentation said it is better to use one big single store, that means many people probably have a very big store with lots of updates, but usually only save a smaller part of that big store to the persistent storage. This means there are lots of unnecessarily storage updates. 
Zustand internally already knows if something has changed after calling the global `set()`, so Zustand should use that to determine if a call to the setItem is necessary or not.

**Comment by dai-shi** [comment]
Sounds like a reasonable improvement.

**Reply by Fabioni** [reply]
So should I move it to an issue?

**Reply by dai-shi** [reply]
This is fine as a bug report. Would you like to work on it?

**Reply by Fabioni** [reply]
Actually no, I just added a check in my custom storage whether the value has changed or not before calling the actual function. That's good enough for me. However, I still think it would be a useful improvement to the condition.

**Reply by dai-shi** [reply]
That's a reasonable solution. I would probably do the same.

If someone wants to work on it, feel free.
